wxcore/Setup.hs

```diff
@@ -39,21 +39,9 @@ wxcoreDirectoryQuoted  = "\"" ++ wxcoreDirectory ++ "\""
 
 -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
 
--- |This slightly dubious function obtains the install path for the wxc package we are using.
--- It works by finding the wxc package's installation info, then finding the include directory 
--- which contains wxc's headers (amongst the wxWidgets include dirs) and then going up a level.
--- It would be nice if the path was part of InstalledPackageInfo, but it isn't.
+--TODO don't hardcode this
 wxcInstallDir :: LocalBuildInfo -> IO FilePath
-wxcInstallDir lbi = 
-    case searchByName (installedPkgs lbi) "wxc" of
-        Unambiguous (wxc_pkg:_) -> do
-            wxc <- filterM (doesFileExist . (</> "wxc.h")) (includeDirs wxc_pkg)
-            case wxc of
-                [wxcIncludeDir] -> return (takeDirectory wxcIncludeDir)
-                [] -> error "wxcInstallDir: couldn't find wxc include dir"
-                _  -> error "wxcInstallDir: I'm confused. I see more than one wxc include directory from the same package"
-        Unambiguous [] -> error "wxcInstallDir: Cabal says wxc is installed but gives no package info for it"
-        _ -> error "wxcInstallDir: Couldn't find wxc package in installed packages"
+wxcInstallDir lbi = pure "/usr/include/wxc/"

```

Comment:
This needs fixing.

Reply:
Maybe something like https://github.com/pranaysashank/wxHaskell/commit/f78d79eb869019731f8ed839f603003d4050137e#diff-26b6d3b6d64b3d830c56815fb3f6d4d88043dbe0d3336eb3fe807ad91d3eda81R44.

But that presumably only works because the path is hardcoded in the cabal file instead.

Reply:
One option we have is to set the build type to Configure and populate the fields in wxcore.buildinfo.in similar to how they do it for X11 Haskell bindings [X11.buildinfo.in](https://github.com/xmonad/X11/blob/master/X11.buildinfo.in)

Reply:
That sounds promising! Not fully sure how it works though

Reply:
Since it's a native library what about using pkg-config (through pkgconfig-depends) instead?

Reply:
> Since it's a native library what about using pkg-config (through pkgconfig-depends) instead?

This would be optimal, but e.g. on Debian the developer package of wx does not ship a .pc file:

~~~~
$ dpkg --listfiles libwxgtk3.0-gtk3-dev | fgrep pkg
~~~~

Reply:
I mean for wxc (see the commented line)

Reply:
FWIW, I think the question of how to distribute `wxc` is the only real blocker on this PR. I've been intending for a while to try packaging it for AUR and/or Nix. But really, I've just been hoping that someone who knows more about building C libraries comes along and offers to work on it.

Reply:
I have a draft of a nix package, I can add it to nixpkgs.

If I try to build wxcore with it though I get this error:

```
src/haskell/Graphics/UI/WXCore/WxcTypes.hs:123:2: error:
     fatal error: wxc/wxc_def.h: No such file or directory
      123 | 
          |  ^              
    |
123 | 
    |  ^
```

Reply:
Duh, there's wxc/generate-version-header.sh. Let's see if it works now...

Reply:
Prerequisite for nix packaging: https://github.com/georgefst/wxHaskell/pull/1

Reply:
> I have a draft of a nix package

Could you please share this?

Reply:
[Here you go](https://github.com/fgaz/nixpkgs/blob/wxc/init/pkgs/development/libraries/wxc/default.nix). I'd like to implement [this](https://github.com/georgefst/wxHaskell/pull/2#discussion_r1204312467) before submitting it to nixpkgs though.

Maybe In the meantime I can add it to the wxHaskell repo

Reply:
https://github.com/georgefst/wxHaskell/pull/5